### PR TITLE
move duplicate code to a single private method

### PR DESCRIPTION
**Pull request by ashmaroli**
Just keeping things DRY..

### CODE REVIEW

**Comment by ashmaroli** [comment]
> What do you think of maybe `regenerate_file` instead?

I don't like that since the primary calling is to render.. How about `render_regenerated`..?

**Comment by Crunch09** [comment]
@ashmaroli 👍

File: lib/jekyll/site.rb [hunk 1]
```diff
@@ -449,21 +449,23 @@ def configure_file_read_opts
...
-          page.output = Jekyll::Renderer.new(self, page, payload).run
-          page.trigger_hooks(:post_render)
-        end
+        render_regenerated(page, payload)
+      end
+    end
+
+    private
+    def render_regenerated(document, payload)
+      if regenerator.regenerate?(document)
```
Comment (parkr) [inline lib/jekyll/site.rb]: Let's use a guard clause here: 

```ruby
return unless regenerator.regenerate?(document)

document.output = ...
````

**Comment by oe** [comment]
@jekyllbot: merge +dev

**Pull request merged**